src/nbl/video/ILogicalDevice.cpp

```diff
@@ -33,18 +33,43 @@ bool ILogicalDevice::updateDescriptorSets(uint32_t descriptorWriteCount, const I
 {
     for (auto i = 0; i < descriptorWriteCount; ++i)
     {
-        auto* ds = static_cast<IGPUDescriptorSet*>(pDescriptorWrites[i].dstSet);
-        if (!ds->processWrite(pDescriptorWrites[i]))
+        const auto& write = pDescriptorWrites[i];
+        auto* ds = static_cast<IGPUDescriptorSet*>(write.dstSet);
+
+        if (!ds->getLayout()->isCompatibleDevicewise(ds))
+            return false;
```

Comment:
this is obvious, you can pretty much assert it, you wouldn't have been able to create a Descriptor Set on a pool without a Layout that comes from the same device

You're getting the Layout from the DS itself, it must be valid by construction.

Reply:
`assert`-ed. Resolved.